channeldb/graph.go

```diff
@@ -190,6 +190,39 @@ func (c *ChannelGraph) Database() *DB {
 	return c.db
 }
 
+// NumNodes returns the number of nodes currently in the graph
+func (c *ChannelGraph) NumNodes(tx *bbolt.Tx) (int, error) {
+	var result int
+	var err error
+
+	count := func(tx *bbolt.Tx) error {
+		nodes := tx.Bucket(nodeBucket)
+		if nodes == nil {
+			return ErrGraphNotFound
+		}
+
+		return nodes.ForEach(func(pubKey, _ []byte) error {
+			if len(pubKey) == 33 {
+				result++
+			}
+
+			return nil
+		})
+	}
+
+	if tx != nil {
+		err = count(tx)
+	} else {
+		err = c.db.View(count)
```

Comment:
is this codepath ever taken?

Reply:
no, I follow the pattern in other methods in this file. In any case, when I update to use a hardcoded node count this code will be removed.